tests/integration/test_transformers.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from rdt import HyperTransformer
+from rdt.transformers import BaseTransformer
+from tests.datasets import BaseDatasetGenerator
+
+DATA_SIZE = 1000
+TEST_COL = 'test_col'
+
+# Mapping of rdt data type to dtype
+DATA_TYPE_TO_DTYPES = {
+    'boolean': ['b', 'O'],
+    'categorical': ['O', 'i', 'f'],
+    'datetime': ['M'],
+    'numerical': ['f', 'i'],
+    'integer': ['i'],
+    'float': ['f', 'i'],
+}
+
+
+def _build_generator_map():
+    """Build a map of data type to data generator.
+
+    Output:
+        dict:
+            A mapping of data type (str) to a list of data
+            generators (rdt.tests.datasets.BaseDatasetGenerator).
+    """
+    generators = {}
+
+    for g in BaseDatasetGenerator.__subclasses__():
+        generators[g.DATA_TYPE] = g.__subclasses__()
+
+    return generators
+
+
+def _find_dataset_generators(data_type, generators):
+    """Find the dataset generators for the given data_type."""
+    return generators.get(data_type, [])
+
+
+def _validate_input_type(input_type):
+    """Check that the transformer input type is not null."""
+    assert input_type is not None
+
+
+def _validate_transformed_data(transformer, transformed_data):
+    """Check that the transformed data is the expected dtype."""
+    expected_data_types = transformer.get_output_types()
+    transformed_dtypes = transformed_data.dtypes
+
+    for column, expected_data_type in expected_data_types.items():
+        assert column in transformed_data
+        assert transformed_dtypes[column].kind in DATA_TYPE_TO_DTYPES[expected_data_type]
+
+
+def _validate_reverse_transformed_data(transformer, reversed_data, input_dtype):
+    """Check that the reverse transformed data is the expected dtype.
+
+    Expect that the dtype is equal to the dtype of the input data.
+    """
+    expected_data_type = transformer.get_input_type()
+    assert reversed_data.dtypes[TEST_COL].kind in DATA_TYPE_TO_DTYPES[expected_data_type]
+
+
+def _validate_composition(transformer, reversed_data, input_data):
+    """Check that the reverse transformed data is equal to the input.
+
+    This is only applicable if the transformer has the composition
+    identity property.
+    """
+    if not transformer.is_composition_identity():
+        return
+
+    if isinstance(reversed_data, pd.DataFrame):
+        reversed_data = reversed_data[TEST_COL]
+    if isinstance(reversed_data, pd.DatetimeIndex):
```

Comment:
I think these should be `elif`

Reply:
Multiple of these conditions could apply - for example, if the reversed_data is a DataFrame and a datetime. I could `elif` the two `isinstance` checks.